Editor/Scripts/Server/EditorServer.cs

```diff
@@ -1,31 +1,58 @@
 using System;
-using System.Threading;
 using UnityEditor;
 using UnityEngine;
 using Object = UnityEngine.Object;
 
 namespace Unity.MeshSync.Editor {
 
 [InitializeOnLoad]
-internal static class AutoSetup {
+internal static class EditorServer {
 
-    private static Server m_server;
+    private static Server            m_server;
+
+    private const string CONFIGURATION_TIP =
+        "You can configure the editor server in Edit/ProjectSettings/MeshSync/EditorServer";
+
+    static EditorServer() {
+        ApplySettingsIfDirty();
+    }
     
-    static AutoSetup() {
-        // Create server
-        var settings = ServerSettings.defaultValue;
+    internal static void ApplySettingsIfDirty() {
         
-        settings.port = AutoSetupSettings.instance.Port;
+        if (!EditorServerSettings.instance.Dirty)
+            return;
         
-        if (!Server.Start(ref settings, out m_server)) {
-            Debug.LogError("[MeshSync] Could not start Server");
+        ApplySettings();
+        
+        EditorServerSettings.instance.Dirty = false;
+    }
+
+    private static void ApplySettings() {
+        
+        EditorApplication.update -= UpdateCall;
+        m_server.Stop();
+        
+        if (!EditorServerSettings.instance.Active) {
+            Debug.Log("[MeshSync] Stopping Editor Server.\n" + CONFIGURATION_TIP);
+            
             return;
         }
         
-        //Register calls
-        EditorApplication.update -= UpdateCall;
         EditorApplication.update += UpdateCall;
+        
+        var settings = ServerSettings.defaultValue;
+        settings.port = EditorServerSettings.instance.Port;
+        
+        
+        
+    
+        if (!Server.Start(ref settings, out m_server)) {
```

Comment:
our target users generally don't see logs.
I think it's better to show a window.